src/components/LifeEventPicker.tsx

```diff
@@ -1,54 +1,287 @@
-import React from 'react';
+import React, { useState } from 'react';
 import TextField from '@material-ui/core/TextField';
 import Grid from '@material-ui/core/Grid';
 import Typography from '@material-ui/core/Typography';
+import Link from '@material-ui/core/Link';
 import { makeStyles } from '@material-ui/core/styles';
 
 export default function LifeEventPicker(): JSX.Element {
   const classes = useStyles();
 
+  const [state, setState] = useState({
```

Comment:
It's more common to call `useState` once for each field. That makes the code to toggle them much cleaner and less error-prone. Something like this:
```
const [birth, setBirth] = useState(true);
const [residence, setResidence] = useState(false);
...

function toggleBirthTextField(): void {
  setBirth(!birth);
}
// Or showBirthTextField(show: boolean): void {...}
```

Of if you'd like, the code can be even simpler since it's just a call to `setBirth`:
```
const [birth, setBirth] = useState(true);
...
<Link onClick={() => setBirth(!birth)} .../>
```

Reply:
Made the changes thank you! That was an embarrassing mistake ^^;